waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/mapping/service/impl/StaticDatabaseMappingServiceTest.java

```diff
@@ -458,6 +459,25 @@ public void panopticOperationsHandlerGetAllDatabases() {
     assertThat(handler.getAllDatabases(), is(allDatabases));
   }
 
+  @Test
+  public void panopticOperationsHandlerGetPrimaryAllDatabases() throws Exception {
+    String pattern = "*";
+    List<String> allPrimaryDatabases =
+        Lists.newArrayList("primary_db1", "primary_db2", "primary_db3");
+    when(primaryDatabaseClient.get_databases(pattern)).thenReturn(allPrimaryDatabases);
+    when(primaryDatabaseClient.get_all_databases()).thenReturn(allPrimaryDatabases);
+
+    primaryMetastore.setMappedDatabases(Lists.newArrayList("primary_db1"));
+
+    service = new StaticDatabaseMappingService(metaStoreMappingFactory,
+        Arrays.asList(primaryMetastore, federatedMetastore), queryMapping);
+
+
+    StaticDatabaseMappingPanopticOperationHandler handler =
+        (StaticDatabaseMappingPanopticOperationHandler) service.getPanopticOperationHandler();
+    assertThat(handler.getPrimaryAllDatabases(), is(Lists.newArrayList("primary_db1")));
```

Comment:
This asserts that the code you added work, but it doesn't explain to me what the behaviour is that you want to change.

Please write a test that tests the difference in behaviour you are after?

It might help if you explain which e.g. hive queries  or HMS methods you're trying to call that behave incorrectly?

Reply:
I have write a test now, pls review .

> This asserts that the code you added work, but it doesn't explain to me what the behaviour is that you want to change.
> 
> Please write a test that tests the difference in behaviour that you like to implement?
> 
> It might help if you explain which e.g. hive queries or HMS methods you're trying to call that behave incorrectly?